TeapotTests/DeliveryQueueTests.swift

```diff
@@ -0,0 +1,189 @@
+
+        let dispatchQueue = DispatchQueue(label: "dispatch.queue.here")
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!,
+                            deliveryQueue: dispatchQueue)
+
+        teapot.put("/put") { result in
+            XCTAssertTrue(!Thread.isMainThread)
+            let queueLabel = String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)!
+
+            XCTAssertEqual(dispatchQueue.label, queueLabel)
+
+            expectation.fulfill()
+        }
+
+        self.waitForExpectations(timeout: 20.0)
+    }
+
+    func testDeliveryOnQueueDelete() {
+        let expectation = self.expectation(description: "Delete")
+
+        let dispatchQueue = DispatchQueue(label: "dispatch.queue.here")
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!,
+                            deliveryQueue: dispatchQueue)
+
+        teapot.delete("/delete") { result in
+            XCTAssertTrue(!Thread.isMainThread)
+            let queueLabel = String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)!
+
+            XCTAssertEqual(dispatchQueue.label, queueLabel)
+
+            expectation.fulfill()
+        }
+
+        self.waitForExpectations(timeout: 20.0)
+    }
+
+    func testDeliveryOnQueueImage() {
+        let expectation = self.expectation(description: "Image")
+
+        let dispatchQueue = DispatchQueue(label: "dispatch.queue.here")
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!,
+                            deliveryQueue: dispatchQueue)
+
+        teapot.downloadImage { result in
+            XCTAssertTrue(!Thread.isMainThread)
+
+            let queueLabel = String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)!
+            XCTAssertEqual(dispatchQueue.label, queueLabel)
+
+            expectation.fulfill()
+        }
+
+        self.waitForExpectations(timeout: 20.0)
+    }
+    
+    func testDeliveryOnMain() {
+        let expectation = self.expectation(description: "Get")
+
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!)
+
+        teapot.get("/get") { result in
+            XCTAssertTrue(Thread.isMainThread)
+            expectation.fulfill()
+        }
+
+        self.waitForExpectations(timeout: 20.0)
+    }
+
+    func testDeliveryQueueOverrideGet() {
+        let expectation = self.expectation(description: "Override Get")
+        let dispatchQueue = DispatchQueue(label: "dispatch.queue.overriden")
+        let teapot = Teapot(baseURL: URL(string: "https://httpbin.org")!)
+
+        teapot.get("/get", deliveryQueue: dispatchQueue) { result in
+            XCTAssertTrue(!Thread.isMainThread)
+
+            let queueLabel = String(cString: __dispatch_queue_get_label(nil), encoding: .utf8)!
+            XCTAssertEqual(dispatchQueue.label, queueLabel)
+
+            expectation.fulfill()
+        }
+
+        self.waitForExpectations(timeout: 20.0)
```

Comment:
worth adding scenarios where you validate that if you *don't* override the delivery queue, it *does* come back on main?

Reply:
👍

Reply:
Wait, like on :107?

Reply:
Ah i see that, must have missed it since it was only on `get` instead of on everything. Thanks!